src/format/kdbx4/parse.rs

```diff
@@ -68,6 +68,9 @@ pub(crate) fn decrypt_kdbx4(
     // parse header
     let (outer_header, inner_header_start) = parse_outer_header(data)?;
 
+    #[cfg(feature = "challenge_response")]
+    let db_key = db_key.clone().perform_challenge(&outer_header.kdf_seed)?;
```

Comment:
I'm not a fan of having to clone the key here, but I don't see another solution if we want to keep passing the key as a reference.